benchmarking/analysis.go

```diff
@@ -0,0 +1,319 @@
+package benchmarking
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+
+	benchmarkingtypes "github.com/LimeChain/gosemble/primitives/benchmarking"
+)
+
+var (
+	errZeroBenchmarkResults = errors.New("provided benchmark results must be more than 0.")
+)
+
+// todo other analysis types
+// todo enum for analysis types
+// todo all enum values to implement Analysis interface
+// todo implement benchmark flag for analysis choice
+// type analysisType int
+
+// const (
+// 	MinSquares analysisType = iota
+// 	MedianSlopes
+// )
+
+// func (a analysisType) Analysis(benchmarkResults []benchmarkResult) (extrinsicTime, reads, writes analysis, err error) {
+// 	if len(benchmarkResults) == 0 {
+// 		err = errZeroBenchmarkResults
+// 		return
+// 	}
+
+// 	if len(benchmarkResults[0].components) == 1 {
+// 		return medianValuesAnalysis(benchmarkResults)
+// 	}
+
+// 	switch a {
+// 	case MinSquares:
+// 		return minSquaresAnalysis(benchmarkResults)
+// 	case MedianSlopes:
+// 		return medianSlopesAnalysis(benchmarkResults)
+// 	default:
+// 		return
+// 	}
+// }
```

Comment:
Do we need these? 
If the current analysis implementation is sufficient, we don't need different implementations right now